### PR TITLE
Update sample apps to AGP 8.1.1

### PR DESCRIPTION
**Pull request by AbdelrahmanTalat**
This PR updates the sample apps to user AGP version 8.1.1 as well as addresses the following issues 

- Add coreLibraryDesugaring to support APIs below 26
- Fix the gradient drawable angle 
- Fix unescaped letter in strings.xml
- Use the support library to ensure that we can render the gradient on lower APIs 
- Remove all the example testing code from the repo
- Update Zendesk dependency to version 1.14.1

### CODE REVIEW

**Comment by IanArb** [comment]
Hey @Z3nAjo could we not considering consolidating all the demo apps into a single app module? It seems a bit redundant having sample demo apps per functionality. We could use a navigation bar of some sort to navigate to each demo functionality. In this case, it reduces our maintenance going forward.

**Review by Melveena-Jolly — APPROVED** [review]

**Comment by ahmedadeltito** [comment]
> Hey @Z3nAjo could we not considering consolidating all the demo apps into a single app module? It seems a bit redundant having sample demo apps per functionality. We could use a navigation bar of some sort to navigate to each demo functionality. In this case, it reduces our maintenance going forward.

Hey, @IanArb, I think @Z3nAjo has a point to separate them to make it easier for the Advocate team to trace the features and target the clients to what project does what feature, but I am still not sure.

File: zendesk_sdk_jwt_demo/app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
     <string name="app_name">Zendesk JWT Demo App</string>
     <string name="app_name_toolbar">Zendesk Demo App</string>
-    <string name="channel_key">{Add your channel key here}</string>
+    <string name="channel_key">eyJzZXR0aW5nc191cmwiOiJodHRwczovL3o0bm5tdGVzdGFwcC56ZW5kZXNrLmNvbS9tb2JpbGVfc2RrX2FwaS9zZXR0aW5ncy8wMUVXSlRWNzdaWjNZUFpRVkRWNE5XMlc5Uy5qc29uIn0=</string>
```
Comment (ahmedadeltito) [inline zendesk_sdk_jwt_demo/app/src/main/res/values/strings.xml:5]: Should we remove the channel key from here? @AbdelrahmanTalat

File: zendesk_sdk_push_demo/app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
     <string name="app_name">Zendesk Demo Push App</string>
     <string name="app_name_toolbar">Zendesk Demo App</string>
-    <string name="channel_key">{Add your channel key here}</string>
+    <string name="channel_key">eyJzZXR0aW5nc191cmwiOiJodHRwczovL3o0bm5tdGVzdGFwcC56ZW5kZXNrLmNvbS9tb2JpbGVfc2RrX2FwaS9zZXR0aW5ncy8wMUVXSlRWNzdaWjNZUFpRVkRWNE5XMlc5Uy5qc29uIn0=</string>
```
Comment (ahmedadeltito) [inline zendesk_sdk_push_demo/app/src/main/res/values/strings.xml:5]: Should we remove the channel key from here? @AbdelrahmanTalat

**Pull request merged**